On branch Lesson_3

SVM71 opened the pull request:
Changes to be committed:
	new file:   Lesson_3/GB_posts_parcer.py
	new file:   Lesson_3/database.py
	new file:   Lesson_3/models.py

SVM71 committed:
On branch Lesson_3
 Changes to be committed:
	new file:   Lesson_3/GB_posts_parcer.py
	new file:   Lesson_3/database.py
	new file:   Lesson_3/models.py

djo1006 commented on Lesson_3/GB_posts_parcer.py:
```diff
@@ -0,0 +1,171 @@
+    @staticmethod
+    def get_date(date_in: str):
+        tmp_lst = date_in.split(' ')
+        u_date = dt.date(year=int(tmp_lst[2]), month=c_user_date._month_dict[tmp_lst[1][:3]], day=int(tmp_lst[0]))
```
на сайте дата время в iso формате, есть намного проще метод преобразовать - datetime.fromisoformat(soup.time.attrs['datetime'])

djo1006 commented on Lesson_3/GB_posts_parcer.py:
```diff
@@ -0,0 +1,171 @@
+
+    ### комментарии в виде (автор комментария и текст комментария)
+    #'https://geekbrains.ru/api/v2/comments?commentable_type=Post&commentable_id=35&order=desc'
+    comment_params = { 'commentable_type': '', 'commentable_id':0, 'order':'' }
```
комментарии надо доставать из api - json

djo1006 commented on Lesson_3/GB_posts_parcer.py:
```diff
@@ -0,0 +1,171 @@
+params = { 'page' : 62 }
+url_p = parse.urlparse(url)
+
+resp = requests.get(url, headers=headers, params=params)
```
не оч понятно как обходятся все страницы

djo1006 commented on Lesson_3/models.py:
```diff
@@ -0,0 +1,40 @@
+    Column('post_id', Integer, ForeignKey('t_posts.id')),
+    Column('tag_id', Integer, ForeignKey('t_tag.id'))
+)
+
```
нет таблицы для комментариев

SVM71 commented:
Спасибо, вроде все по делу!) Есть недоработки. А комменты вроде из jsonа я вытаскивал. 13:52, 26 октября 2020 г., djo1006 <notifications@github.com>:
@djo1006 commented on this pull request.



In Lesson_3/models.py:
> +from sqlalchemy import (
+    Column,
+    Integer,
+    String,
+    ForeignKey,
+    Table
+)
+
+Base = declarative_base()
+
+t_teg_post = Table(
+    't_teg_post', Base.metadata,
+    Column('post_id', Integer, ForeignKey('t_posts.id')),
+    Column('tag_id', Integer, ForeignKey('t_tag.id'))
+)
+

нет таблицы для комментариев

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
Отправлено из мобильной Яндекс.Почты: http://m.ya.ru/ymail

djo1006 reviewed (commented):
> Спасибо, вроде все по делу!) Есть недоработки. А комменты вроде из jsonа я вытаскивал. 13:52, 26 октября 2020 г., djo1006 <notifications@github.com>: @djo1006 commented on this pull request. In Lesson_3/models.py:
> +from sqlalchemy import (
> + Column, + Integer, + String, + ForeignKey, + Table +) + +Base = declarative_base() + +t_teg_post = Table( + 't_teg_post', Base.metadata, + Column('post_id', Integer, ForeignKey('t_posts.id')), + Column('tag_id', Integer, ForeignKey('t_tag.id')) +) + нет таблицы для комментариев —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe. Отправлено из мобильной Яндекс.Почты: http://m.ya.ru/ymail

Сори по комментам недосмотрел, все ок.

SVM71 committed:
On branch Lesson_3
 Changes to be committed:
	new file:   Lesson_3/GB_parser.py
	modified:   Lesson_3/GB_posts_parcer.py
	modified:   Lesson_3/database.py
	modified:   Lesson_3/models.py
